frappe/public/js/frappe/utils/common.js

```diff
@@ -332,10 +332,37 @@ frappe.utils.sanitise_redirect = (url) => {
 		};
 	})();
 
+	/*
+	* Strips out url containing the text `javascript` with or without any HTML Entities in it
+	**/
 	const sanitise_javascript = (url) => {
-		// please do not ask how or why
+		/*
+		* Written below split into parts, but actual is in one line regardless of whitespaces
+		* /
+		* 	j
+		* 		\s*(&#x.{1,7})?
+		* 	a
+		* 		\s*(&#x.{1,7})?
+		* 	v
+		* 		\s*(&#x.{1,7})?
+		* 	a
+		* 		\s*(&#x.{1,7})?
+		* 	s
+		* 		\s*(&#x.{1,7})?
+		* 	c
+		* 		\s*(&#x.{1,7})?
+		* 	r
+		* 		\s*(&#x.{1,7})?
+		* 	i
+		* 		\s*(&#x.{1,7})?
+		* 	p
+		* 		\s*(&#x.{1,7})?
+		* 	t
+		* /gi
+		* */
+		const REGEX_ESC_UNIT = /\s*(&#x.{1,7})?/;
```

Comment:
With exception of change for this whitespace part it seems identical:

```diff
- `[\s]*`
+ \s*
```

Reply:
TBH I still don't understand what this does :woozy_face:

Reply:
> TBH I still don't understand what this does 🥴

It strips out the xss in the format `javascript:alert('Gotchu!')` -> `alert('Gotchu!')` possibly feom the url.